tests/positive/Termination/Data/Nat.juvix

```diff
@@ -1,7 +1,7 @@
 module Data.Nat;
 
-syntax fixity add {arity: binary, assoc: left};
-syntax fixity mul {arity: binary, assoc: left, above: [add]};
+syntax fixity add := binary {assoc := left};
+syntax fixity mul := binary {assoc := left; above := [add]};
```

Comment:
Not specific to this example, but two concerns here:
Shouldn't the formatter refine the above list? e.g:

```diff
-syntax fixity mul := binary {assoc := left; above := [add;X;add]};
+syntax fixity mul := binary {assoc := left; above := [add;X]};
```

In case there is field duplication entry, e.g.:

```
syntax fixity seq := binary {assoc := left; assoc:=right};
```
 The error should say that explicitly, instead of:

```
X.juvix:6:57:
  |
6 | syntax fixity seq := binary {assoc := left; assoc:=right};
  |                                                         ^
expecting above, below, or same
```

Reply:
The simplest way to parse this is by using [Applciative permutation](https://hackage.haskell.org/package/parser-combinators-1.3.0/docs/Control-Applicative-Permutations.html) + megaparsec. That's what I'm currently doing, so I don't have fine control over error messages. If we really wanted custom errors, we would need to parse this with a more general parser and check for repeated fields. In my opinion, this is not worth the necessary effort and extra code.